do/droplets.go

```diff
@@ -174,3 +184,24 @@ func (i *instances) dropletByName(ctx context.Context, nodeName types.NodeName)
 
 	return nil, cloudprovider.InstanceNotFound
 }
+
+// dropletIDFromProviderID returns a droplet's ID extracted from the node's
+// providerID spec. The providerID spec should be retrievable from the Kubernetes
+// node object. The expected format is: digitalocean://droplet-id
+func dropletIDFromProviderID(providerID string) (string, error) {
```

Comment:
There are some comments below that are just nitpicks. Take them or not.

Also consider: is there any reason to split on the slashes at all? Why not just 
```
func dropletIDFromProviderID(providerID string) (string, error) {
    const scheme := providerID + "://"

    if !strings.HasPrefix(providerID, scheme) != 0 {
        return "", fmt.Errorf("unexpected providerID format: %s, format should be: digitalocean://12345", providerID)
    }
    return strings.TrimPrefix(providerID, scheme), nil
}
```

Reply:
Did you mean 
```
const scheme := providerName + "://"
```
?

Reply:
yes

Reply:
That's a great reason for using `splitN` or `HasPrefix`, though. Right now on master, `instances.NodeAddressesByProviderID` will look for the `providerID` given. All this PR is about is making sure that `digitalocean://` is a prefix for that. The PR, though, changes behavio where previously `instances.dropletById` would have been given `//droplet-id`.

Unless we're prepared to validate other rules about the `droplet-id` value, not allowing leading `/`s seems out of scope.

Reply:
Having said all that, this is all a nitpick; I don't intend it to be a blocker.